PicPi/AddPhotoViewController.swift

```diff
@@ -0,0 +1,196 @@
+    super.loadView()
+    view.backgroundColor = .white
+    margins = view.layoutMarginsGuide
+    setupSendButton()
+    setupDoneBtn()
+    setupTableView()
+    photoUploadManager.delegate = self
+    
+  }
+  
+  // MARK: - UI Setup
+  private func setupSendButton() {
+    
+    view.addSubview(sendButton)
+    sendButton.addTarget(self, action: #selector(sendButtonPressed), for: .touchUpInside)
+    sendButton.setTitle("Send", for: .normal)
+    sendButton.setTitleColor(.darkGray, for: .disabled)
+    sendButton.isEnabled = false
+    sendButton.sizeToFit()
+    sendButton.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate(
+      [
+        sendButton.centerXAnchor.constraint(equalTo: margins.centerXAnchor) ,
+        sendButton.centerYAnchor.constraint(equalTo : margins.bottomAnchor, constant:  -sendButton.frame.height)  ,
+        sendButton.widthAnchor.constraint(equalToConstant: 100)
+      ]
+    )
+    
+  }
+  private func setupDoneBtn() {
+    
+    view.addSubview(doneButton)
+    doneButton.addTarget(self, action: #selector(doneButtonPressed), for: .touchUpInside)
+    doneButton.setTitle("Done", for: .normal)
+    doneButton.sizeToFit()
+    doneButton.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate(
+      [
+        doneButton.leftAnchor.constraint(equalTo: margins.leftAnchor ),
+        doneButton.centerYAnchor.constraint(equalTo : margins.topAnchor, constant:  doneButton.frame.height)  ,
+        doneButton.widthAnchor.constraint(equalToConstant: 100)  ]
+    )
+    
+  }
+  private func setupTableView() {
+    tableView.allowsMultipleSelection = false
+    tableView.dataSource = self
+    tableView.delegate = self
+    view.addSubview(tableView)
+    tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+    tableView.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate(
+      [
+        tableView.topAnchor.constraint(equalTo: doneButton.bottomAnchor, constant: 10),
+        tableView.leftAnchor.constraint(equalTo: view.leftAnchor),
+        tableView.rightAnchor.constraint(equalTo: view.rightAnchor),
+        tableView.bottomAnchor.constraint(equalTo: sendButton.topAnchor),
+      ]
+    )
+  }
+  
+  // MARK: - Buttons Actions
+  @objc func sendButtonPressed(sender: UIButton!) {
+    photoUploadManager.startUploadPreviouslySelectedPhotos()
+    sendButton.isEnabled = false
+    
+  }
+  @objc func doneButtonPressed(sender: UIButton!) {
+    self.dismiss(animated: true, completion: nil)
+  }
+  
+}
+// MARK: - BSImagePicker Setup
+extension AddPhotoViewController {
+  
+  func showImagePickerController(){
+    
+    let imagePicker = ImagePickerController()
+    imagePicker.modalPresentationStyle = .fullScreen
+    presentImagePicker(imagePicker, select: { (asset) in
+      
+    }, deselect: { (asset) in
+      
+    }, cancel: { (assets) in
+      
+    }, finish: { [self] (assets) in
```

Comment:
Since this is UI code and not some background task, no need to capture self weakly

Remove `[self]` since this is the same as `strong self`

Reply:
I meant to use weak self ,but since that is not needed I'll remove it , 
Thanks for the advice